packages/malloy/src/lang/test/expressions.spec.ts

```diff
@@ -1116,6 +1116,54 @@ describe('expressions', () => {
     expect(expr`${name} = NULL`).toTranslate();
   });
 });
+describe('alternations as in', () => {
+  test('a=b|c', () => {
+    expect('ai=1|2').compilesTo('{ai in {1},{2}}');
+  });
+  test('a!=b|c', () => {
+    expect('ai!=1|2').compilesTo('{ai not in {1},{2}}');
+  });
+  test('a=(b|c)', () => {
+    expect('ai=(1|2)').compilesTo('{ai in {1},{2}}');
+  });
+  test('a?b|c', () => {
+    expect('ai?1|2').compilesTo('{ai in {1},{2}}');
+  });
+  test('a=(b)|c', () => {
+    expect('ai=(1)|2').compilesTo('{ai in {1},{2}}');
+  });
+  test('a=b|c|d', () => {
+    expect('ai=1|2|3').compilesTo('{ai in {1},{2},{3}}');
+  });
+  test('a=(b|c)|d', () => {
+    expect('ai=(1|2)|3').compilesTo('{ai in {1},{2},{3}}');
+  });
+  test('a=b|(c|d)', () => {
+    expect('ai=1|(2|3)').compilesTo('{ai in {1},{2},{3}}');
+  });
+  test('a=b|c&d', () => {
+    expect('ai=1|2&3').compilesTo('{{ai = 1} or {{ai = 2} and {ai = 3}}}');
+  });
+  test('a=b|>d', () => {
+    expect('ai=1|>2').compilesTo('{{ai = 1} or {ai > 2}}');
+    expect(expr`ai=1|>2`).toLog(
+      warningMessage(
+        'Only | seperated values are legal when used with = operator'
+      )
+    );
+  });
+  test('a ? = (b | c)', () => {
+    expect('ai ? (= 1 | 2)').compilesTo('{ai in {1},{2}}');
+  });
+  test('legacy in', () => {
+    const inExpr = expr`ai in (1,2,3)`;
+    expect(inExpr).compilesTo('{ai in {1},{2},{3}}');
+    expect(inExpr).toLog(warningMessage('Use = (a|b|c) instead of IN (a,b,c)'));
+  });
+  // mtoy todo this hits "alternation tree has no value"
+  test.todo('a ? (= 1) | 2');
+});
+describe('rigor around ? and =', () => {});
```

Comment:
Are you planning on filling this in?

Reply:
resolved as a long standing bug in our grammar, noted this and left it skipped